tests/rptest/tests/schema_registry_test.py

```diff
@@ -1223,6 +1224,118 @@ def move_leader():
                 check_connection(n.account.hostname)
             move_leader()
 
+    @cluster(num_nodes=3)
+    def test_restart_schema_registry(self):
+        # Create several schemas on one subject and return the list schemas and schema ids
+        def register_schemas(subject: str):
+            schemas = [
+                json.dumps({"schema": schema1_def}),
+                json.dumps({"schema": schema2_def}),
+                json.dumps({"schema": schema3_def})
+            ]
+            schema_ids = []
+
+            for schema in schemas:
+                self.logger.debug(f"Post schema {json.loads(schema)}")
+                result_raw = self._post_subjects_subject_versions(
+                    subject=subject, data=schema)
+                self.logger.debug(result_raw)
+                assert result_raw.status_code == requests.codes.ok
+                res = result_raw.json()
+                self.logger.debug(res)
+                assert "id" in res
+                schema_ids.append(result_raw.json()["id"])
+
+            return schemas, schema_ids
+
+        # Given a list of subjects, check that they exist on the registry
+        def check_subjects(subjects: list[str]):
+            result_raw = self._get_subjects()
+            self.logger.debug(result_raw)
+            assert result_raw.status_code == requests.codes.ok
+            res_subjects = result_raw.json()
+            self.logger.debug(res_subjects)
+            assert type(res_subjects) == type([])
+            res_subjects.sort()
+            subjects.sort()
+            assert res_subjects == subjects
+
+        # Given the subject and list of versions, check that the version numbers match
+        def check_subject_versions(subject: str, subject_versions: list[int]):
+            result_raw = self._get_subjects_subject_versions(subject=subject)
+            self.logger.debug(result_raw)
+            assert result_raw.status_code == requests.codes.ok
+            res_versions = result_raw.json()
+            self.logger.debug(res_versions)
+            assert type(res_versions) == type([])
+            res_versions.sort()
+            subject_versions.sort()
```

Comment:
nit: I think these should always be in order.

Reply:
Is that always guaranteed though?

Reply:
I think so. The version should be monotonically increasing as its written to the _schemas topic. And it should be read in order and applied to the store in order.

I don't actually recall what happens on a hard delete of the last version of subject; it may use the next version, it may use the deleted version. It may depend on whether a restart occurred. It may depend on whether compaction occurred after the delete.

Reply:
> I think so. The version should be monotonically increasing as its written to the _schemas topic. And it should be read in order and applied to the store in order.

In that case, I should not call sort then because if `assert res_version == subject_version` fails, then that is actually a bug and not a test failure. I'll fix in the followup pr

Reply:
Done in https://github.com/redpanda-data/redpanda/pull/8127/commits/f0417002c0a8d5772987cb92f736fc3543f94375